pkg/util/deployment/deployment.go

```diff
@@ -132,19 +133,44 @@ func GetReplicaCountForRCs(replicationControllers []*api.ReplicationController)
 }
 
 // Returns the number of available pods corresponding to the given RCs.
-func GetAvailablePodsForRCs(c client.Interface, rcs []*api.ReplicationController) (int, error) {
-	// TODO: Use MinReadySeconds once https://github.com/kubernetes/kubernetes/pull/12894 is merged.
+func GetAvailablePodsForRCs(c client.Interface, rcs []*api.ReplicationController, minReadySeconds int) (int, error) {
 	allPods, err := getPodsForRCs(c, rcs)
 	if err != nil {
 		return 0, err
 	}
+	return getReadyPodsCount(allPods, minReadySeconds), nil
+}
+
+func getReadyPodsCount(pods []api.Pod, minReadySeconds int) int {
 	readyPodCount := 0
-	for _, pod := range allPods {
+	for _, pod := range pods {
 		if api.IsPodReady(&pod) {
-			readyPodCount++
+			// Check if we've passed minReadySeconds since LastTransitionTime
+			// If so, this pod is ready
+			readyConditions := 0
+			validReadyConditions := 0
+			for _, c := range pod.Status.Conditions {
+				// we only care about pod ready conditions
+				if c.Type == api.PodReady {
```

Comment:
There shouldnt be more than 1 ready condition, right?
We can break as soon as we find a ready condition.
We wont need readyCondition and validReadyConditions will then be a boolean instead of int.